test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/AppTests.kt

```diff
@@ -16,49 +16,38 @@
 
 package io.realm.test.mongodb.shared
 
-import io.realm.internal.platform.singleThreadDispatcher
 import io.realm.mongodb.App
-import io.realm.mongodb.AppConfiguration
 import io.realm.mongodb.EmailPassword
-import io.realm.mongodb.appConfigurationOf
-import io.realm.mongodb.internal.KtorNetworkTransport
+import io.realm.test.mongodb.TestApp
+import io.realm.test.mongodb.admin.ServerAdmin
 import kotlinx.coroutines.runBlocking
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
 import kotlin.test.Ignore
 import kotlin.test.Test
 
-const val TEST_APP_1 = "testapp1" // Id for the default test app
-const val BASE_URL = "http://127.0.0.1:9090"
-
 // Cannot run on CI yet, as it requires sync server to be started with
-// tools/sync_test_server/start_server.sh and manual creation of a user "asdf@asdf.com"/"asdfasdf"
-// through the web ui
+// tools/sync_test_server/start_server.sh
 @Ignore
 class AppTests {
 
-    @Test
-    fun login() {
-        // Send request directly to the local server to get the actual app ID
-        // TODO Wrap test app setup as in Realm Java
-        //  https://github.com/realm/realm-kotlin/pull/447#discussion_r707350138
-        val applicationId = KtorNetworkTransport(
-            timeoutMs = 5000,
-            dispatcher = singleThreadDispatcher("transport dispatcher")
-        ).sendRequest(
-            "get",
-            "http://127.0.0.1:8888/$TEST_APP_1",
-            mapOf(),
-            "",
-            true
-        ).let { response ->
-            when (response.httpResponseCode) {
-                200 -> response.body
-                else -> throw IllegalStateException(response.toString())
-            }
-        }
+    lateinit var app: App
+    lateinit var admin: ServerAdmin
 
-        val configuration: AppConfiguration = appConfigurationOf(applicationId, BASE_URL, singleThreadDispatcher("asdf"))
-        val app = App.create(configuration)
+    @BeforeTest
+    fun setup() {
+        app = TestApp()
+        admin = ServerAdmin(app)
+        admin.createUser("asdf@asdf.com", "asdfasdf")
+    }
 
+    @AfterTest
+    fun teaddown() {
+        admin.deleteAllUsers()
```

Comment:
In Java, this is part of an App.close() extension method.  I suspect we need to add this eventually here as well. We might as well just do it in this PR as it should be a trivial amount of code.

Reply:
I added an `asTestApp` extension method and bundled all the `AdminApp` implementations up on `TestApp`, so it becomes 
https://github.com/realm/realm-kotlin/pull/453/files#diff-7789c96c2ad7327e4d0f561d78b70e1cee7f84b15af77df4bf1852da42d686f0R35

to avoid polluting the actual `App` api with extension methods for test.